modules/core/src/shaderlib/project/project.glsl.js

```diff
@@ -32,42 +32,43 @@ uniform float project_uCoordinateSystem;
 uniform float project_uScale;
 uniform bool project_uWrapLongitude;
 uniform float project_uAntimeridian;
-uniform vec3 project_uPixelsPerMeter;
-uniform vec3 project_uPixelsPerDegree;
-uniform vec3 project_uPixelsPerUnit;
-uniform vec3 project_uPixelsPerUnit2;
+uniform vec3 project_uDistancePerMeter;
+uniform vec3 project_uDistancePerDegree;
+uniform vec3 project_uDistancePerUnit;
+uniform vec3 project_uDistancePerUnit2;
+uniform float project_uPixelsPerDistance;
 uniform vec4 project_uCenter;
 uniform mat4 project_uModelMatrix;
 uniform mat4 project_uViewProjectionMatrix;
 uniform vec2 project_uViewportSize;
 uniform float project_uDevicePixelRatio;
 uniform float project_uFocalDistance;
 uniform vec3 project_uCameraPosition;
-uniform vec2 project_coordinate_origin;
+uniform vec2 project_uCoordinateOrigin;
 
 const float TILE_SIZE = 512.0;
 const float PI = 3.1415926536;
 const float WORLD_SCALE = TILE_SIZE / (PI * 2.0);
 const vec2 ZERO_64_XY_LOW = vec2(0.0, 0.0);
 
 //
-// Scaling offsets - scales meters to "pixels"
-// Note the scalar version of project_scale is for scaling the z component only
+// Scaling offsets - scales meters to "world distance"
+// Note the scalar version of project_size is for scaling the z component only
 //
-float project_scale(float meters) {
-  return meters * project_uPixelsPerMeter.z;
+float project_size(float meters) {
+  return meters * project_uDistancePerMeter.z;
 }
 
-vec2 project_scale(vec2 meters) {
-  return meters * project_uPixelsPerMeter.xy;
+vec2 project_size(vec2 meters) {
+  return meters * project_uDistancePerMeter.xy;
 }
 
-vec3 project_scale(vec3 meters) {
-  return meters * project_uPixelsPerMeter;
+vec3 project_size(vec3 meters) {
+  return meters * project_uDistancePerMeter;
 }
 
-vec4 project_scale(vec4 meters) {
-  return vec4(meters.xyz * project_uPixelsPerMeter, meters.w);
+vec4 project_size(vec4 meters) {
```

Comment:
In the past we have added `project_scale(..) {return project_size() }` stubs for backwards compatibility. 

Given that upgrading custom layers tends to be biggest pain point by far for apps, should we do it this time as well?

The warnings system would give apps one major release to deal with this in orderly fashion.